resultsdbupdater/utils.py

```diff
@@ -182,66 +182,67 @@ def ci_metrics_post_to_resultsdb(msg):
     msg_body = msg['body']['msg']
-    tests = msg_body['tests']
-
-    arch = msg_body['environment']['arch']
-    brew_tag = msg_body['environment']['brew_tag']
-    build_type = msg_body['environment']['build_type']
-
-    jenkins_job_url = msg_body['infrastructure']['jenkins_job_url']
-    jenkins_build_url = msg_body['infrastructure']['jenkins_build_url']
-
-    cips_report = msg_body['results']['cips_report']
-    cips_status = msg_body['results']['cips_status']
-
-    testcase_url = jenkins_job_url
+    msg_body_ci = msg['body']['msg']['ci']
+    msg_body_artifact = msg['body']['msg']['artifact']
+    url = msg_body['run']['url']
+    outcome = msg_body['status']
+    category = msg_body['category']
+    issuer = msg_body_artifact['issuer']
+    rebuild = msg_body['run']['rebuild']
+    log = msg_body['run']['log']
+    system_os = msg_body['system'][0]['os']
+    system_provider = msg_body['system'][0]['provider']
+    ci_name = msg_body_ci['name']
+    ci_url = msg_body_ci['url']
+    ci_environment = msg_body_ci['environment']
+    ci_team = msg_body_ci['team']
+    ci_irc = msg_body_ci['irc']
+    ci_email = msg_body_ci['email']
 
     # variables to be passed to create_result
     groups = [{
         'uuid': str(uuid.uuid4()),
-        'ref_url': jenkins_build_url
+        'url': url
     }]
-    outcome = cips_status
-    ref_url = jenkins_build_url
+    outcome = outcome
+    url = url
 
     result_data = {
         'item': component,
-        'ci_type': ci_type,
-        'brew_task_id': brew_task_id,
-        'brew_tag': brew_tag,
-        'arch': arch,
+        'type': type,
+        'id': id,
+        'category': category,
         'component': component,
-        'build_type': build_type,
-        'cips_report': cips_report,
-        'cips_status': cips_status,
-        'testcase_url': testcase_url,
+        'scratch': scratch,
+        'status': outcome,
+        'issuer': issuer,
+        'rebuild': rebuild,
+        'log': log,
+        'system_os': system_os,
+        'system_provider': system_provider,
+        'ci_name': ci_name,
+        'ci_url': ci_url,
+        'ci_environment': ci_environment,
+        'ci_team': ci_team,
+        'ci_irc': ci_irc,
+        'ci_email': ci_email
     }
 
-    # Create individual test results for each test in the message
-    for test_name, test_outcome in tests.items():
-        testcase = {
-            'name': 'rpm-factory.cips.{}'.format(test_name),
-            'ref_url': testcase_url
-        }
-        if not create_result(
-                session, testcase, test_outcome, ref_url, result_data, groups):
-            LOGGER.error(
-                'A new result for message "{0}" couldn\'t be created'
-                .format(msg_id))
-            return False
-
-    # Create the overall test result
+    # # Create the overall test result
     testcase = {
-        'name': 'rpm-factory.cips',
-        'ref_url': testcase_url
+        'name': 'cips',
+        'ref_url': url
```

Comment:
The testcase URL should point to documentation about what CIPS is or at least the Jenkins URL to the CIPS project. The testcase is supposed to be a very generic thing that encompasses all tests of the same type.

Reply:
Ok